pkg/federation/model/actor.ts

```diff
@@ -0,0 +1,223 @@
+import type { AccountID } from '../../accounts/model/account.js';
+import type { ID } from '../../id/type.js';
+import type { ActorKeyPair } from './actorKey.js';
+import type { InstanceID } from './instance.js';
+
+export type ActorID = ID<Actor>;
+
+export interface CreateActorArgs {
+  id: ActorID;
+  instanceID: InstanceID;
+  accountID: AccountID;
+  profileURL: URL;
+
+  inboxURL: URL;
+  outboxURL: URL;
+  followersURL: URL;
+  followingURL: URL;
+  sharedInboxURL: URL;
+  actorKeyPair: ActorKeyPair[];
+}
+
+/**
+ * ActivityPub Actor
+ * @see https://www.w3.org/TR/activitypub/#actors
+ */
+export class Actor {
+  /**
+   * Actor ID
+   * @example 31415926535
+   */
+  private readonly id: ActorID;
+  /**
+   * InstanceID
+   * @example 27182818284
+   */
+  private readonly instanceID: InstanceID;
+  /**
+   * Account ID
+   * @example 141421356
+   */
+  private readonly accountID: AccountID;
+  /**
+   * Actor profile URL
+   * @example `https://social.example.com/accounts/@john@example.com`
+   */
+  private readonly profileURL: URL;
+
+  /**
+   * Actor Inbox URL
+   * @example `https://social.example.com/accounts/31415926535/inbox`
+   */
+  private inboxURL: URL;
+  /**
+   * Actor Outbox URL
+   * @example `https://social.example.com/accounts/31415926535/outbox`
+   */
+  private outboxURL: URL;
+  /**
+   * Actor Followers URL
+   * @example `https://social.example.com/accounts/31415926535/followers`
+   */
+  private followersURL: URL;
+  /**
+   * Actor following URL
+   * @example `https://social.example.com/accounts/31415926535/following`
+   */
+  private followingURL: URL;
+  /**
+   * Actor SharedInbox URL
+   * @see https://www.w3.org/TR/activitypub/#shared-inbox-delivery
+   * @example `https://social.example.com/shared`
+   */
+  private sharedInboxURL: URL;
+
+  /**
+   * Actor KeyPairs for activity signing / verifying
+   * NOTE: Actor has multiple key pairs for activity signing / verifying
+   * @see https://codeberg.org/fediverse/fep/src/branch/main/fep/521a/fep-521a.md
+   * @see https://w3c.github.io/vc-data-integrity/vocab/security/vocabulary.html#Multikey
+   */
+  private readonly actorKeyPair: ActorKeyPair[];
```

Comment:
```suggestion
  private readonly actorKeyPair: readonly ActorKeyPair[];
```
書き換えない想定でしたら, `readonly` を配列オブジェクトの方にも付与してください.